Firefox Selection Fix.sh

```diff
@@ -6,8 +6,23 @@
 return_wd="$PWD"
 reason_already_root='already_root'
 description='The Firefox Selection Fix script disables the broken clickSelectsAll behavior of Firefox. Make sure Firefox is up-to-date and closed'
-firefox_dir=$(whereis firefox | cut -d ' ' -f 2)
-fallback_firefox_dir='/usr/lib/firefox' # Fallback path: put your Firefox install path here. The install path includes the `firefox` binary and a `browser` directory.
+
+### Find out the operating system
+os='other'
+if [ $(uname -s) = 'Darwin' ]; then os='macos'; fi
+
+# Find Firefox's inner directory
+if [ $os = 'macos' ]; then
+  # Most probable place:
+  firefox_dir='/Applications/Firefox.app/Contents/Resources'
+  # … unless it's elsewhere:
+  fallback_firefox_dir=$( mdfind -name 'Firefox.app' | head -n 1 )'/Contents/Resources'
+else
+  firefox_dir=$(whereis firefox | cut -d ' ' -f 2)
+  # Fallback path: put your Firefox install path here. The install path includes the `firefox` binary and a `browser` directory.
+  fallback_firefox_dir='/usr/lib/firefox'
+fi
```

Comment:
You put a fixed path in `firefox_dir`, but the automatically determined path in `fallback_firefox_dir`, which is exactly the other way around in the (original) `else` branch. These variables are flipped, semantically.

The original comment gives clear guidance as to how to edit the shell script in case the directory isn’t found, but it’s only part of this conditional branch. We actually don’t need two separate `fallback_firefox_dir` assignments. One suffices for both operating systems (outside of the `if`).

```suggestion
if [ "$os" = 'macos' ]; then
  firefox_dir="$( mdfind -name 'Firefox.app' | head -n 1 )/Contents/Resources"
else
  firefox_dir=$(whereis firefox | cut -d ' ' -f 2)
fi

# Fallback path: put your Firefox install path here. The install path includes the `firefox` binary and a `browser` directory.
# Possible fallback for MacOS: /Applications/Firefox.app/Contents/Resources
fallback_firefox_dir='/usr/lib/firefox'
```